File: src/core/functions/reverseBuffer.ts
function reverseBuffer(input: Uint8Array, keyArray: Uint8Array, initor:any): Uint8Array {
    const length = input.length;
    const reversed = new Uint8Array(length);

    for (let i = 0; i < length; i++) {
        reversed[i] = input[length - 1 - i];
    }

    return reversed;
}

let funPair = {
    obfuscation: reverseBuffer,
    deobfuscation: reverseBuffer
}

export default funPair;

File: android/wireguard_obfuscation/obfuscationFuns/ReverseBuffer.java
package com.morph_vpn.plugin.wireguard_obfuscation.obfuscationFuns;
import java.util.Arrays;

public class ReverseBuffer {
    public static byte[] reverseBuffer(byte[] input, byte[] keyArray, Object initor) {
        byte[] reversed = new byte[input.length];

        for (int i = 0; i < input.length; i++) {
            reversed[i] = input[input.length - 1 - i];
        }

        return reversed;
    }

    public static void main(String[] args) {
        byte[] input = new byte[]{/* initialize input byte array */};
        byte[] keyArray = new byte[]{/* initialize keyArray byte array */};
        Object initor = null;

        byte[] reversed = reverseBuffer(input, keyArray, initor);

        System.out.println("Input: " + Arrays.toString(input));
        System.out.println("Reversed: " + Arrays.toString(reversed));
    }
}
